bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ThingStatusInfoI18nLocalizationService.java

```diff
@@ -87,7 +87,7 @@ protected void setTranslationProvider(TranslationProvider i18nProvider) {
     }
 
     protected void unsetTranslationProvider(TranslationProvider i18nProvider) {
-        i18nProvider = null;
+        this.i18nProvider = null;
```

Comment:
You are right, the variable names has been not replaced and so they are not really correct.
But this does not introduce that bug at all.